source/common/upstream/upstream_impl.h

```diff
@@ -620,6 +625,10 @@ class ClusterInfoImpl : public ClusterInfo {
   const bool warm_hosts_;
   absl::optional<std::string> eds_service_name_;
   const absl::optional<envoy::api::v2::Cluster::CustomClusterType> cluster_type_;
+
+  class FactoryContextImpl;
```

Comment:
?

Reply:
Oops, that's a left over from the original PR. Removed the forward declaration.